From e3128dee26346969d3ff99844b7f04affdd5ab62 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Tue, 24 Mar 2015 19:23:38 +0100
Subject: [PATCH 7/9] Allow to skip the check for Python headers

When building without _giscanner.la (e.g., during the second step of a
MinGW cross compilation) there is no need to build the Python module,
hence there is no need to check for its headers (that in turn would
require the installation of the whole python2 toolchain on the host).
---
 configure.ac | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index a6a272d..2c6cf94 100644
--- a/configure.ac
+++ b/configure.ac
@@ -272,9 +272,12 @@ case "$host" in
 	pyexecdir=`echo $pyexecdir | tr '\\\\' '/'`
 	;;
 esac
-AM_CHECK_PYTHON_HEADERS(, AC_MSG_ERROR([Python headers not found]))
-if test "x$os_win32" = "xyes"; then
-  AM_CHECK_PYTHON_LIBS(, AC_MSG_ERROR([Python libs not found. Windows requires Python modules to be explicitly linked to libpython.]))
+AC_ARG_ENABLE(giscanner,AS_HELP_STRING([--disable-giscanner],[disable giscanner python module]),,enable_giscanner=yes)
+if test "x$enable_giscanner" != "xno"; then
+  AM_CHECK_PYTHON_HEADERS(, AC_MSG_ERROR([Python headers not found]))
+  if test "x$os_win32" = "xyes"; then
+    AM_CHECK_PYTHON_LIBS(, AC_MSG_ERROR([Python libs not found. Windows requires Python modules to be explicitly linked to libpython.]))
+  fi
 fi
 
 dnl Not enabled by default until 3.6 cycle when we can propose mako as
-- 
2.7.1

